redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -25,133 +25,144 @@ import app.cash.redwood.Modifier
 
-        val widget = items.itemForGlobalIndex(cellForItemAtIndexPath.item.toInt())
-        cell.set(widget.value)
+  override val items: Widget.Children<UIView> = processor.items
 
-        return cell
-      }
-    }
+  private val dataSource = object : NSObject(), UITableViewDataSourceProtocol {
+    override fun tableView(
+      tableView: UITableView,
+      numberOfRowsInSection: NSInteger,
+    ) = processor.size.toLong()
 
-  private val collectionViewDelegate: UICollectionViewDelegateFlowLayoutProtocol =
-    object : NSObject(), UICollectionViewDelegateFlowLayoutProtocol {
-      override fun collectionView(
-        collectionView: UICollectionView,
-        layout: UICollectionViewLayout,
-        minimumLineSpacingForSectionAtIndex: NSInteger,
-      ): CGFloat {
-        return 0.0
-      }
+    override fun tableView(
+      tableView: UITableView,
+      cellForRowAtIndexPath: NSIndexPath,
+    ) = processor.getCell(cellForRowAtIndexPath.item.toInt())
+  }
 
+  private val tableViewDelegate: UITableViewDelegateProtocol =
+    object : NSObject(), UITableViewDelegateProtocol {
       override fun scrollViewDidScroll(scrollView: UIScrollView) {
-        val visibleIndexPaths = collectionView.indexPathsForVisibleItems()
+        val visibleIndexPaths = tableView.indexPathsForVisibleRows ?: return
 
         if (visibleIndexPaths.isNotEmpty()) {
           // TODO: Optimize this for less operations
-          updateViewport(
+          onViewportChanged?.invoke(
             visibleIndexPaths.minOf { (it as NSIndexPath).item.toInt() },
             visibleIndexPaths.maxOf { (it as NSIndexPath).item.toInt() },
           )
         }
       }
     }
 
-  override val placeholder = MutableListChildren<UIView>()
-
   init {
-    collectionView.apply {
-      dataSource = collectionViewDataSource
-      delegate = collectionViewDelegate
+    tableView.apply {
+      dataSource = this@UIViewLazyList.dataSource
+      delegate = tableViewDelegate
       prefetchingEnabled = true
+      rowHeight = UITableViewAutomaticDimension
+      estimatedRowHeight = 44.0
+      separatorStyle = UITableViewCellSeparatorStyleNone
 
       registerClass(
-        LazyListContainerCell(CGRectZero.readValue()).classForCoder() as ObjCClass?,
-        reuseIdentifier,
+        cellClass = LazyListContainerCell(UITableViewCellStyle.UITableViewCellStyleDefault, reuseIdentifier)
+          .initWithFrame(CGRectZero.readValue()).classForCoder() as ObjCClass?,
+        forCellReuseIdentifier = reuseIdentifier,
       )
     }
   }
 
+  final override fun onViewportChanged(onViewportChanged: (Int, Int) -> Unit) {
+    this.onViewportChanged = onViewportChanged
+  }
+
   override fun isVertical(isVertical: Boolean) {
-    collectionViewFlowLayout.scrollDirection = if (isVertical) {
-      UICollectionViewScrollDirectionVertical
-    } else {
-      UICollectionViewScrollDirectionHorizontal
-    }
+    // TODO: support horizontal LazyLists.
```

Comment:
Do we plan to support this in the future, or can/should we punt this indefinitely? (Has this already been built on the Android side?)

Reply:
I’d like to see what happens if I swap out UITableView for UICollectionView. If I’m lucky it’ll just be tracking slight differences in the APIs with no performance impact. I think I’ll be lucky.

Reply:
@dnagler To answer your question directly:

1. We plan to support this in the future. We'll need this within Cash for things like the QAB, but also, `LazyRow` should be provided for external consumers (i.e., non Cash people), as these components are supposed to be the lazy counterparts to the existing `Column`/`Row` components.
2. Yes, it's already been built on the Android side.